src/inmanta/server/services/orchestrationservice.py

```diff
@@ -517,6 +624,38 @@ async def _put_version(
         *,
         connection: asyncpg.connection.Connection,
     ) -> None:
+        """
+        :param rid_to_resource: This parameter should contain all the resources when a full compile is done.
+                                When a partial compile is done, it should contain all the resources that belong to the
+                                updated resource sets or the shared resource sets.
```

Comment:
The object is mutated by this method, may be useful to mention that?
```suggestion
                                updated resource sets or the shared resource sets.
                                This method updates this object with purge-on-delete resources.
```